client.go

```diff
@@ -1123,9 +1122,6 @@ func (c *Client) PerformRequest(method, path string, params url.Values, body int
 // Optionally, a list of HTTP error codes to ignore can be passed.
 // This is necessary for services that expect e.g. HTTP status 404 as a
 // valid outcome (Exists, IndicesExists, IndicesTypeExists).
-//
-// If ctx is not nil, it uses the ctxhttp to do the request,
-// enabling both request cancelation as well as timeout.
 func (c *Client) PerformRequestC(ctx context.Context, method, path string, params url.Values, body interface{}, ignoreErrors ...int) (*Response, error) {
```

Comment:
Did you double-check that no code calls `PerformRequestC` with `ctx` set to `nil`?

Reply:
Seems like you didn't?! https://travis-ci.org/olivere/elastic/jobs/287568213#L751

Reply:
I think a possible solution would be to check for `ctx == nil` in `PerformRequestC` and initialize it with `context.Background()` then.